api/client/create.go

```diff
@@ -77,7 +77,7 @@ func newCIDFile(path string) (*cidFile, error) {
 	return &cidFile{path: path, file: f}, nil
 }
 
-func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runconfig.HostConfig, cidfile, name string) (*types.ContainerCreateResponse, error) {
+func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runconfig.HostConfig, cidfile, name string, pull bool) (*types.ContainerCreateResponse, error) {
```

Comment:
`pull` is just a client only flag, I think it does not make sense to add to `Config` struct